### PR TITLE
Add auth to the masking endpoints

### PR DESCRIPTION
**Pull request by ThomasLaPiana**
Closes  https://github.com/ethyca/security-issues/issues/13

### Code Changes

* [x] rename `verify_oauth_client_cli` -> `verify_oauth_client_prod`
* [x] add `MASKING` scopes for `MASKING:EXEC` and `MASKING:READ`
* [x] add the new `MASKING` scopes to the `VIEWER` role (they are low-risk and without side-effects)
* [x] add the `Security` dependency to the `masking` endpoints

### Steps to Confirm

* [ ] in a `prod` security environment, try to hit the `masking` endpoints, expect a 401
* [ ] after authenticating (logging in), try again and see it work as expected

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [x] Issue Requirements are Met
* [x] Update `CHANGELOG.md`
* [ ] ~~Relevant Follow-Up Issues Created~~
* [ ] ~~For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated~~

### Description Of Changes

This is PR adds authentication to the `masking` endpoints as well as accompanying `scopes`. Note that it is only authenticated in `prod` mode to make it a non-breaking change.

### CODE REVIEW

**Review by RobertKeyser — COMMENTED** [review]
Looks good from my end. No concerns, but others should probably give the explicit approval :)

File: src/fides/lib/oauth/roles.py [hunk 1]
```diff
@@ -94,6 +96,8 @@ class RoleRegistryEnum(Enum):
     DATA_SUBJECT_READ,
     DATA_USE_READ,
     EVALUATION_READ,
+    MASKING_EXEC,
```
Comment (ThomasLaPiana) [inline src/fides/lib/oauth/roles.py:99]: I added these to viewer scopes, but please check me on if this makes sense! Does the masking endpoint just return a masked value or does it go ahead and mask something with a side effect?

Comment (pattisdr) [reply src/fides/lib/oauth/roles.py:99]: Masking endpoint just returns a masked value - no side effects in the system.  It's a weird one, this endpoint. I keep changing my mind as to whether a "viewer" should have this scope or not. I mean it doesn't hurt anything. Just not an action I associate with "viewer"

Comment (rsilvery) [reply src/fides/lib/oauth/roles.py:99]: Fine with me! I think it's unlikely they'll know about this :)

**Pull request merged**